### PR TITLE
don't filter available commands in the extension

### LINKED ISSUE #236099: Terminal suggest: Don't do any filtering of builtins/globals on the extension side
Looks like filtering it happening here: https://github.com/microsoft/vscode/blob/c99c6e2e9262c37b32d887a719e7a987fdea75ea/extensions/terminal-suggest/src/terminalSuggestMain.ts#L312-L319

This filtering should be done exclusively on the renderer size so we can backspace without needing to hide the widget, as well as properly fuzzy matching items.

### PR DESCRIPTION
**Pull request by meganrogge**
fix #236099

### CODE REVIEW

File: extensions/terminal-suggest/src/terminalSuggestMain.ts [hunk 1]
```diff
@@ -159,10 +160,13 @@ function getLabel(spec: Fig.Spec | Fig.Arg | Fig.Suggestion | string): string[]
 }
 
 function createCompletionItem(commandLine: string, cursorPosition: number, prefix: string, label: string, description?: string, kind?: vscode.TerminalCompletionItemKind): vscode.TerminalCompletionItem {
```
Comment (meganrogge) [inline extensions/terminal-suggest/src/terminalSuggestMain.ts:162]: follows same logic as `TerminalCompletionService`, which now works very well.

File: extensions/terminal-suggest/src/terminalSuggestMain.ts [hunk 2]
```diff
@@ -95,6 +95,7 @@ export async function activate(context: vscode.ExtensionContext) {
 				// const cwd = resolveCwdFromPrefix(prefix, terminal.shellIntegration?.cwd) ?? terminal.shellIntegration?.cwd;
 				return new vscode.TerminalCompletionList(result.items, { filesRequested: result.filesRequested, foldersRequested: result.foldersRequested, cwd: result.cwd, pathSeparator: osIsWindows() ? '\\' : '/' });
 			}
+			console.log(result.items.map(r => r.label + ' ' + r.replacementIndex).join('\n'));
```
Comment (meganrogge) [inline extensions/terminal-suggest/src/terminalSuggestMain.ts:98]:
```suggestion
```

File: extensions/terminal-suggest/src/terminalSuggestMain.ts [hunk 3]
```diff
@@ -329,8 +330,9 @@ export async function getCompletionItemsFromSpecs(specs: Fig.Spec[], terminalCon
 
 	if (!specificSuggestionsProvided && (filesRequested === foldersRequested)) {
 		// Include builitin/available commands in the results
+		const labels = new Set(items.map(i => i.label));
```
Comment (Tyriar) [inline extensions/terminal-suggest/src/terminalSuggestMain.ts:333]: We might want an approach that actually merges them eventually here, the ones we provide here might end up being richer. For example the pwsh ones I was working on will have descriptions as provided by pwsh.

File: extensions/terminal-suggest/src/terminalSuggestMain.ts [hunk 4]
```diff
@@ -329,8 +330,9 @@ export async function getCompletionItemsFromSpecs(specs: Fig.Spec[], terminalCon
 
 	if (!specificSuggestionsProvided && (filesRequested === foldersRequested)) {
```
Comment (Tyriar) [inline extensions/terminal-suggest/src/terminalSuggestMain.ts:331]: I think `specificSuggestionsProvided` is badly named here? Does it actually mean an argument was provided? We want availableCommands to get filled whenever the command is being input, I think that condition would be:

```ts
commandLine.substring(0, cursorIndex).trimStart().includes(' ')
```

File: extensions/terminal-suggest/src/test/terminalSuggestMain.test.ts [hunk 5]
```diff
@@ -56,7 +56,6 @@ function createCodeTestSpecs(executable: string): ITestSpec2[] {
 		{ input: `${executable} |`, expectedCompletions: codeOptions },
 		{ input: `${executable} --locale |`, expectedCompletions: localeOptions },
 		{ input: `${executable} --diff |`, expectedResourceRequests: { type: 'files', cwd: testCwd } },
-		{ input: `${executable} -di|`, expectedCompletions: codeOptions.filter(o => o.startsWith('di')), expectedResourceRequests: { type: 'both', cwd: testCwd } },
```
Comment (Tyriar) [inline extensions/terminal-suggest/src/test/terminalSuggestMain.test.ts:59]: Shouldn't this still be passing? `-di` should fuzzy match `--diff`

**Review by Tyriar — DISMISSED** [review]

File: extensions/terminal-suggest/src/test/terminalSuggestMain.test.ts (hunk 5 above)
Comment (meganrogge) [reply extensions/terminal-suggest/src/test/terminalSuggestMain.test.ts:59]: Yeah and it does. For some reason, the test always passes locally and not in CI.

Comment (meganrogge) [reply extensions/terminal-suggest/src/test/terminalSuggestMain.test.ts:59]: Screenshot 2024-12-16 at 1 13 51 PM

**Review by rebornix — DISMISSED** [review]

**Review by joshspicer — APPROVED** [review]

**Pull request merged**